src/test/java/api/support/fakes/processors/StorageRecordPreProcessors.java

```diff
@@ -45,29 +42,16 @@ private StorageRecordPreProcessors() {
   public static CompletableFuture<JsonObject> setEffectiveLocationIdForItem(
     @SuppressWarnings("unused") JsonObject oldItem, JsonObject newItem) {
 
-    String permanentLocationId = newItem.getString(ItemProperties.PERMANENT_LOCATION_ID);
-    String temporaryLocationId = newItem.getString(ItemProperties.TEMPORARY_LOCATION_ID);
-
-    if (ObjectUtils.anyNotNull(temporaryLocationId, permanentLocationId)) {
-      newItem.put(
-        ItemProperties.EFFECTIVE_LOCATION_ID,
-        firstNonNull(temporaryLocationId, permanentLocationId)
-      );
-
-      return CompletableFuture.completedFuture(newItem);
-    }
-
     final String holdingsRecordId = newItem.getString("holdingsRecordId");
-    CompletableFuture<JsonObject> getCompleted = getHoldingById(holdingsRecordId);
+    final JsonObject holding = getHoldingById(holdingsRecordId);
 
-    return getCompleted.thenApply(holding -> {
-      String permanentLocation = holding.getString(ItemProperties.PERMANENT_LOCATION_ID);
-      String temporaryLocation = holding.getString(ItemProperties.TEMPORARY_LOCATION_ID);
+    newItem.put(EFFECTIVE_LOCATION_ID, firstNonNull(
+      newItem.getString(TEMPORARY_LOCATION_ID),
+      newItem.getString(PERMANENT_LOCATION_ID),
+      holding.getString(TEMPORARY_LOCATION_ID),
+      holding.getString(PERMANENT_LOCATION_ID)));
 
-      return newItem.put(ItemProperties.EFFECTIVE_LOCATION_ID,
-        firstNonNull(temporaryLocation, permanentLocation)
-      );
-    });
+    return completedFuture(newItem);
```

Comment:
If there is nothing asynchronous in this code flow, why is a future needed?

Reply:
Will fix it.